src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+import usersFromServer from '../../api/users';
+import { TODO } from '../../types/todo';
+
+interface Props {
+  onAdd: (todo: TODO) => void;
+}
+
+export const NewTodo: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setHasTitleError(false);
+    setHasUserIdError(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setHasTitleError(false);
+    setTitle(event.target.value);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setHasUserIdError(false);
+    setUserId(+event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onAdd({
+      id: 0,
```

Comment:
The `id` field is hardcoded to `0`, which can lead to duplicate IDs. It should be dynamically assigned, possibly by passing the next available ID from the parent component.